lib/tasks/reports.rake

```diff
@@ -0,0 +1,32 @@
+require 'csv'
+require 'date'
+
+namespace :reports do
+  desc 'creates a csv of all submissions for a grant'
+  task :grant_submissions, [:grant] => [:environment] do |t, args|
+    grant = Grant.where(name: args[:grant]).first
+    submissions = grant.submissions
+
+    csv_file = "#{Rails.root}/tmp/#{grant}_submissions_#{Time.now.strftime("%Y-%m-%d-%H-%M-%S")}.csv"
+    puts "Writing submissions to file: #{csv_file}"
+
+    column_headers = [ "applicant last name", "applicant first name", "applicant email", "project title"]
+    questions = []
+    submissions.first.responses.each { |response| questions << response.question.text }
+    column_headers.concat questions
+
+    CSV.open(csv_file, "w") do |csv|
+      csv << column_headers
+
+      submissions.each do |submission|
+        applicant = submission.applicant
+
+        submission_fields = [applicant.last_name, applicant.first_name, applicant.email, submission.title]
+        submission_fields.concat submission.responses.map(&:response_value)
```

Comment:
Simply looping through all responses from submission will not give all responses corresponding to questions. What if some submission question, has null response.

I think here you want to get all responses and then loop through by questions, from top and pull responses for that questions.

Reply:
for e.g

```
questions  = [ q_1, q_2, q_3]
questions.each do |question|
response =  submission.responses.detect {|res| res.question == question}
submission_fields << response.present? ? response.response_value : ""
end
```

Reply:
> Simply looping through all responses from submission will not give all responses corresponding to questions. What if some submission question, has null response.
> 
> I think here you want to get all responses and then loop through by questions, from top and pull responses for that questions.
In addition, submission.responses.map(&:response_value) does not guarantee resoinses come in the same order as your header